leshan-server-core/src/main/java/org/eclipse/leshan/server/client/Client.java

```diff
@@ -62,27 +65,21 @@
 
     private final LinkObject[] objectLinks;
 
+    private Map<String, String> additionalRegistrationAttributes = new HashMap<String, String>();
+
     /** The location where LWM2M objects are hosted on the device */
     private final String rootPath;
 
     private final Date lastUpdate;
 
-    public Client(String registrationId, String endpoint, InetAddress address, int port,
-            InetSocketAddress registrationEndpointAddress) {
-        this(registrationId, endpoint, address, port, null, null, null, null, null, registrationEndpointAddress);
-    }
-
+    @Deprecated
```

Comment:
As we break the API (by adding parameters), I think we can remove the deprecated. 
If we want to force user to use Builder, we can make the constructor protected.